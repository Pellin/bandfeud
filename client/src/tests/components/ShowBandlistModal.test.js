import React from 'react';
import { shallow } from 'enzyme';
import ShowBandListModal from '../../components/highscores/ShowBandlistModal';

let bands, playerInfo, wrapper;

beforeEach(() => {
  bands = [
    { name: 'one', url: 'www.1img.com', points: 12 },
    { name: 'two', url: 'www.2img.com' },
    { name: 'three', url: 'www.3img.com', points: 31 },
    { name: 'four', url: 'www.4img.com' }
  ];
  playerInfo = {
    player: 'Falle',
    date: 1570214169008,
    score: 297
  };
});

it('should render component correctly', () => {
  const onClick = () => {};
  const wrapper = shallow(
    <ShowBandListModal showModal={true} playerInfo={playerInfo} bands={bands} onClick={onClick} />
  );
  expect(wrapper.find('img').length).toBe(4);
  expect(wrapper.find({ alt: bands[2].name }).props()).toMatchObject({
    alt: 'three',
    className: 'band-image',
    src: 'www.3img.com',
    title: 'Open Discogs page for THREE '
  });
  expect(wrapper).toMatchSnapshot();
});

it('should open discogs page when band image is clicked', () => {
  global.open = jest.fn();
  const wrapper = shallow(
    <ShowBandListModal showModal={true} playerInfo={playerInfo} bands={bands} />
  );
  wrapper.find({ alt: bands[0].name }).simulate('click', { button: 0 });
  expect(global.open).toHaveBeenCalledWith(
    `https://www.discogs.com/artist/${bands[0].name}`
  );
});
