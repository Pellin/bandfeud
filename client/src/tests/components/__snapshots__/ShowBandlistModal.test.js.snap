// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="BandList"
  isOpen={true}
  overlayClassName="overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="modal-header"
  >
    <div
      className="header-date"
    >
      04/10/2019
       
    </div>
     
    <div
      className="header-player"
    >
      Falle
      <p>
        2
         bands
      </p>
    </div>
     
    <div
      className="header-points"
    >
      297
      p
    </div>
  </div>
  <div
    className="band-info"
    key="one"
  >
    <div
      className="band-image-container"
    >
      <img
        alt="one"
        className="band-image"
        onClick={[Function]}
        src="www.1img.com"
        title="Open Discogs page for ONE "
      />
    </div>
    <div
      className="band-name-container"
    >
      <div
        className="band-name"
      >
        one
      </div>
    </div>
    <div
      className="band-points"
    >
      12
      p
    </div>
  </div>
  <div
    className="band-info"
    key="two"
  >
    <div
      className="band-image-container"
    >
      <img
        alt="two"
        className="band-image"
        onClick={[Function]}
        src="www.2img.com"
        title="Open Discogs page for TWO "
      />
    </div>
    <div
      className="band-name-container"
    >
      <div
        className="band-name-computer"
      >
        two
      </div>
    </div>
    <div
      className="no-band-points"
    >
      No points
    </div>
  </div>
  <div
    className="band-info"
    key="three"
  >
    <div
      className="band-image-container"
    >
      <img
        alt="three"
        className="band-image"
        onClick={[Function]}
        src="www.3img.com"
        title="Open Discogs page for THREE "
      />
    </div>
    <div
      className="band-name-container"
    >
      <div
        className="band-name"
      >
        three
      </div>
    </div>
    <div
      className="band-points"
    >
      31
      p
    </div>
  </div>
  <div
    className="band-info"
    key="four"
  >
    <div
      className="band-image-container"
    >
      <img
        alt="four"
        className="band-image"
        onClick={[Function]}
        src="www.4img.com"
        title="Open Discogs page for FOUR "
      />
    </div>
    <div
      className="band-name-container"
    >
      <div
        className="band-name-computer"
      >
        four
      </div>
    </div>
    <div
      className="no-band-points"
    >
      No points
    </div>
  </div>
</Modal>
`;
